main/main.cpp

```diff
@@ -1689,12 +1701,22 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 	GLOBAL_DEF("debug/file_logging/log_path", "user://logs/godot.log");
 	GLOBAL_DEF(PropertyInfo(Variant::INT, "debug/file_logging/max_log_files", PROPERTY_HINT_RANGE, "0,20,1,or_greater"), 5);
 
-	if (!project_manager && !editor && FileAccess::get_create_func(FileAccess::ACCESS_USERDATA) &&
-			GLOBAL_GET("debug/file_logging/enable_file_logging")) {
+	// If `--log-file` is used to override the log path, allow creating logs for the project manager or editor
+	// and even if file logging is disabled in the Project Settings.
+	if (FileAccess::get_create_func(FileAccess::ACCESS_USERDATA) &&
+			(!log_file.is_empty() || (!project_manager && !editor && GLOBAL_GET("debug/file_logging/enable_file_logging")))) {
```

Comment:
`--log-file` typically wouldn't be used to write to `user://`, but to a pass relative to the working directory, so I'm not sure the check for `ACCESS_USERDATA` makes sense / is enough.

Reply:
I've tested an absolute path located outside `user://` and it worked fine, but I guess we can check for `ACCESS_FILESYSTEM` when the log path isn't empty.